elytron-oidc-client/subsystem/src/main/java/org/wildfly/extension/elytron/oidc/SecureDeploymentDefinition.java

```diff
@@ -0,0 +1,244 @@
+        for (SimpleAttributeDefinition attribute : ProviderAttributeDefinitions.ATTRIBUTES) {
+            ALL_ATTRIBUTES.add(attribute);
+        }
+    }
+
+    SecureDeploymentDefinition() {
+        super(new Parameters(PathElement.pathElement(ElytronOidcDescriptionConstants.SECURE_DEPLOYMENT),
+                ElytronOidcExtension.getResourceDescriptionResolver(ElytronOidcDescriptionConstants.SECURE_DEPLOYMENT))
+                .setAddHandler(SecureDeploymentAddHandler.INSTANCE)
+                .setRemoveHandler(SecureDeploymentRemoveHandler.INSTANCE)
+                .setAddRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES)
+                .setRemoveRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES));
+    }
+
+    @Override
+    public void registerOperations(ManagementResourceRegistration resourceRegistration) {
+        super.registerOperations(resourceRegistration);
+        resourceRegistration.registerOperationHandler(GenericSubsystemDescribeHandler.DEFINITION, GenericSubsystemDescribeHandler.INSTANCE);
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        super.registerAttributes(resourceRegistration);
+        for (AttributeDefinition attribute : ALL_ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(attribute, null, SecureDeploymentWriteAttributeHandler.INSTANCE);
+        }
+    }
+
+    @Override
+    public void registerChildren(ManagementResourceRegistration resourceRegistration) {
+        resourceRegistration.registerSubModel(new CredentialDefinition());
+        resourceRegistration.registerSubModel(new RedirectRewriteRuleDefinition());
+    }
+
+    static class SecureDeploymentAddHandler extends AbstractAddStepHandler {
+        public static SecureDeploymentAddHandler INSTANCE = new SecureDeploymentAddHandler();
+
+        private SecureDeploymentAddHandler() {
+            super(ALL_ATTRIBUTES);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            super.performRuntime(context, operation, model);
+            String clientId = CLIENT_ID.resolveModelAttribute(context, model).asStringOrNull();
+            String resource = RESOURCE.resolveModelAttribute(context, model).asStringOrNull();
+            if (clientId == null && resource == null) {
+                throw ROOT_LOGGER.resourceOrClientIdMustBeConfigured();
+            }
+
+            boolean disableTrustManager = DISABLE_TRUST_MANAGER.resolveModelAttribute(context, model).asBoolean();
+            if (disableTrustManager) {
+                ROOT_LOGGER.disableTrustManagerSetToTrue();
+            }
+            OidcConfigService oidcConfigService = OidcConfigService.getInstance();
+            oidcConfigService.addSecureDeployment(operation, context.resolveExpressions(model));
```

Comment:
Could you file a Minor JIRA to rework this to use OperationContext.getCurrentAddress()?

A general code style thing we want to do in management code is to get away from parsing addresses out of the operation ModelNode.

Reply:
@bstansberry  Jira issue created: https://issues.redhat.com/browse/WFLY-15292